diff --git a/lib/repoman/copyrights.py b/lib/repoman/copyrights.py
index 275dcbc3f..1fe4309f2 100644
--- a/lib/repoman/copyrights.py
+++ b/lib/repoman/copyrights.py
@@ -15,9 +15,9 @@ from portage import util
 
 
 _copyright_re1 = \
-	re.compile(br'^(# Copyright \d\d\d\d)-\d\d\d\d( Gentoo (Foundation|Authors))\b')
+	re.compile(br'^(# Copyright \d\d\d\d)-\d\d\d\d( ((Gentoo (Foundation|Authors))|Obsidian-Studios))\b')
 _copyright_re2 = \
-	re.compile(br'^(# Copyright )(\d\d\d\d)( Gentoo (Foundation|Authors))\b')
+	re.compile(br'^(# Copyright )(\d\d\d\d)( ((Gentoo (Foundation|Authors))|Obsidian-Studios))\b')
 
 
 class _copyright_repl(object):
@@ -31,7 +31,7 @@ class _copyright_repl(object):
 			return matchobj.group(0)
 		else:
 			return matchobj.group(1) + matchobj.group(2) + \
-				b'-' + self.year + b' Gentoo Authors'
+				b'-' + self.year + matchobj.group(3)
 
 
 def update_copyright_year(year, line):
@@ -51,7 +51,10 @@ def update_copyright_year(year, line):
 	year = _unicode_encode(year)
 	line = _unicode_encode(line)
 
-	line = _copyright_re1.sub(br'\1-' + year + b' Gentoo Authors', line)
+	if b'Obsidian' in line:
+		line = _copyright_re1.sub(br'\1-' + year + b' Obsidian-Studios', line)
+	else:
+		line = _copyright_re1.sub(br'\1-' + year + b' Gentoo Authors', line)
 	line = _copyright_re2.sub(_copyright_repl(year), line)
 	if not is_bytes:
 		line = _unicode_decode(line)
diff --git a/lib/repoman/modules/linechecks/gentoo_header/header.py b/lib/repoman/modules/linechecks/gentoo_header/header.py
index 1f035a97e..2d01bf868 100644
--- a/lib/repoman/modules/linechecks/gentoo_header/header.py
+++ b/lib/repoman/modules/linechecks/gentoo_header/header.py
@@ -42,6 +42,7 @@ class EbuildHeader(LineCheck):
 			if match is None:
 				return self.errors['COPYRIGHT_ERROR']
 			if not (match.group('author') == 'Gentoo Authors' or
+			        match.group('author') == 'Obsidian-Studios, Inc.' or
 					(int(match.group('year2')) < 2019 and
 						match.group('author') == 'Gentoo Foundation')):
 				return self.errors['COPYRIGHT_ERROR']
